# **Korelasi**

Korelasi adalah ukuran statistik yang mengungkapkan sejauh mana dua variabel terkait secara linier artinya mereka berubah bersama pada tingkat yang konstan baik searah maupun berlawanan arah. Korelasi digunakan untuk mendeskripsikan hubungan sederhana tanpa membuat pernyataan tentang sebab dan akibat.

## **Nilai Korelasi**

Koefisien korelasi (biasa disimbolkan dengan *r*) dapat bernilai positif ( + ) jika hubungan dua variabel searah / berbanding lurus (semakin meningkat variabel A maka nilai variabel B juga meningkat pun sebaliknya) atau negatif ( - ) jika hubungan dua variabel berbeda arah (semakin meningkat variabel A maka nilai variabel B akan menurun begitu juga sebaliknya) dan berada pada rentang -1 s.d. 1 mendekati -1 atau 1 maka hubungan keeratan dua variabel semakin kuat. Jika nilainya mendekati 0, maka hubungan keeratan dua variabel semakin lemah.

Berikut ini adalah interpretasi besarnya nilai koefisien korelasi.

*   0.0 <= |*r*| <= 0.2 menyatakan hubungan keeratan sangat lemah
*   0.2 < |*r*| <= 0.4 menyatakan hubungan keeratan lemah
*   0.4 < |*r*| <= 0.7 menyatakan hubungan keeratan cukup kuat
*   0.7 < |*r*| <= 0.9 menyatakan hubungan keeratan kuat
*   0.9 < |*r*| <= 1.0 menyatakan hubungan keeratan sangat kuat.

## **Pearson Correlation**

Korelasi Pearson adalah alat analisis statistik parametrik yang digunakan untuk melihat keeratan hubungan linier antara 2 variabel yang skala datanya adalah interval atau rasio.
<br></br>
<b>Rumus Korelasi Pearson</b>
  \begin{equation}
  r=\frac{\sum_{i=1}^{n}(x_{i} - \bar{x})\cdot(y_{i} - \bar{y})}{\sqrt{\sum_{i=1}^{n}(x_{i} - \bar{x})^2\cdot\sum_{i=1}^{n}(y_{i} - \bar{y})^2}} \\
  \end{equation}

Dengan :  
<li>𝑥 = Variabel pertama </li>
<li>𝑥̄ = Rata-rata variabel pertama </li>
<li>𝑦 = Variabel Kedua</li>
<li>𝑦̄ = Rata-rata variabel kedua </li>
</ol>

## **CONTOH**
Korelasi antara email marketing dan tingkat konversi adalah salah satu aspek yang sangat penting dalam digital marketing. Email marketing digunakan untuk mengirimkan pesan kepada pelanggan atau prospek dengan tujuan mempengaruhi tindakan mereka, seperti melakukan pembelian, mendaftar, atau mengunduh konten. Sementara itu, tingkat konversi mengukur seberapa efektif suatu kampanye dalam mendorong pengunjung untuk melakukan tindakan yang diinginkan, seperti pembelian atau pendaftaran.

Hitung berapa korelasi antara data open_rate dan conversion_rate dibawah ini serta tentukan jenis korelasinya!

In [18]:
# # Data kampanye email
# open_rate = [0.25, 0.3, 0.35, 0.4, 0.45, 0.5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85, 0.9, 0.95]
# conversion_rate = [0.02, 0.03, 0.04, 0.05, 0.06, 0.07, 0.08, 0.09, 0.1, 0.11, 0.12, 0.13, 0.14, 0.15, 0.16]

In [21]:
open_rate = [0.34, 0.57, 0.61, 0.68, 0.41, 0.83, 0.91, 0.26, 0.65, 0.49, 0.74, 0.75, 0.52, 0.85, 0.59] #iklan melalui email
conversion_rate = [0.12, 0.09, 0.14, 0.16, 0.18, 0.06, 0.05, 0.02, 0.1, 0.13, 0.04, 0.03, 0.15, 0.07, 0.08]

**JAWAB :**

1. Cari rata-rata open_rate

In [26]:
rataan_open_rate = sum(open_rate) / len(open_rate)
print(f'Rata-rata open_rate = {rataan_open_rate:.2f}')

Rata-rata open_rate = 0.61


2. Cari rata-rata conversion_rate

In [27]:
rataan_conversion_rate = sum(conversion_rate) / len(conversion_rate)
print(f'Rata-rata conversion_rate = {rataan_conversion_rate:.2f}')

Rata-rata conversion_rate = 0.09


3. Hitung bagian pembilang (dimisalkan sebagai A)

  \begin{equation}
  A = \sum_{i=1}^{n}(x_{i} - \bar{x})\cdot(y_{i} - \bar{y})
  \end{equation}

In [28]:
jumlah_pembilang = 0

for i in range(len(conversion_rate)):
  jumlah_pembilang += (open_rate[i]-rataan_open_rate) * (conversion_rate[i] - rataan_conversion_rate)

print(jumlah_pembilang)

-0.04483333333333333


4. Hitung bagian penyebut (dimisalkan sebagai B)

  \begin{equation}
  B = \sqrt{\sum_{i=1}^{n}(x_{i} - \bar{x})^2\cdot\sum_{i=1}^{n}(y_{i} - \bar{y})^2}
  \end{equation}

In [30]:
import math

jumlah_penyebut_1 = 0
jumlah_penyebut_2 = 0

for i in range(len(open_rate)):
  jumlah_penyebut_1 += ((open_rate[i] - rataan_open_rate)**2)
  jumlah_penyebut_2 += ((conversion_rate[i] - rataan_conversion_rate)**2)

jumlah_penyebut = math.sqrt(jumlah_penyebut_1 * jumlah_penyebut_2)

print(jumlah_penyebut)

0.1328226654018223


5. Lakukan perhitungan korelasi

<b>Rumus Korelasi Pearson</b>
  \begin{equation}
  r=\frac{\sum_{i=1}^{n}(x_{i} - \bar{x})\cdot(y_{i} - \bar{y})}{\sqrt{\sum_{i=1}^{n}(x_{i} - \bar{x})^2\cdot\sum_{i=1}^{n}(y_{i} - \bar{y})^2}} \\
  \end{equation}

In [31]:
r = jumlah_pembilang / jumlah_penyebut

print(f'Korelasi = {r:.4f}')

Korelasi = -0.3375


6. Tentukan jenis korelasi

*   0.0 <= |*r*| <= 0.2 menyatakan hubungan keeratan sangat lemah
*   0.2 < |*r*| <= 0.4 menyatakan hubungan keeratan lemah
*   0.4 < |*r*| <= 0.7 menyatakan hubungan keeratan cukup kuat
*   0.7 < |*r*| <= 0.9 menyatakan hubungan keeratan kuat
*   0.9 < |*r*| <= 1.0 menyatakan hubungan keeratan sangat kuat.

In [32]:
if 0.0 <= abs(r) <= 0.2:
  print('Hubungan keeratan sangat lemah')
elif 0.2 < abs(r) <= 0.4:
  print('Hubungan keeratan lemah')
elif 0.4 < abs(r) <= 0.7:
  print('Hubungan keeratan cukup kuat')
elif 0.7 < abs(r) <= 0.9:
  print('Hubungan keeratan kuat')
else:
  print('Hubungan keeratan sangat kuat')

Hubungan keeratan lemah


In [33]:
import plotly.express as px

fig = px.scatter(
  x = open_rate,
  y = conversion_rate,
  trendline = 'ols',
  color_discrete_sequence = ['#0E2954']
)

fig.show()

## **Interpretasi Hasil**

Koefisien korelasi Pearson -0.3375 menunjukkan bahwa ada korelasi negatif yang lemah antara open rate dan conversion rate. Artinya, secara umum, semakin tinggi tingkat pembukaan email (open rate), semakin rendah tingkat konversi (conversion rate).

## Interpretasi dalam Konteks Pemasaran:

Meskipun tampak berlawanan dengan intuisi, hasil ini bisa terjadi karena beberapa alasan:

* Target audiens yang tidak tepat: Email mungkin dibuka oleh banyak orang, tetapi jika mereka bukan target audiens yang tepat untuk penawaran, tingkat konversi akan rendah.
* Konten email yang tidak relevan: Email mungkin menarik perhatian sehingga dibuka, tetapi isinya tidak relevan atau tidak cukup meyakinkan untuk mendorong konversi.
* Call to action (CTA) yang lemah: Meskipun email dibuka dan dibaca, CTA yang lemah atau tidak jelas dapat menghalangi orang untuk melakukan tindakan yang diinginkan (misalnya, membeli produk, mendaftar, dll.).
* Faktor-faktor lain yang lebih dominan: Mungkin ada faktor lain yang lebih kuat dalam mempengaruhi conversion rate, seperti desain landing page, penawaran produk, harga, dan sebagainya.

## Rekomendasi:

Berdasarkan hasil ini, dapat dicoba beberapa hal berikut:

* Segmentasi audiens yang lebih baik: Pastikan email dikirim ke target audiens yang tepat dan relevan dengan penawaran.
* Optimasi konten email: Pastikan konten email relevan, menarik, dan meyakinkan untuk mendorong konversi.
* Perkuat CTA: Gunakan CTA yang jelas, ringkas, dan menarik perhatian.
* Analisis faktor-faktor lain: Lakukan analisis lebih lanjut untuk mengidentifikasi faktor-faktor lain yang mungkin memengaruhi conversion rate dan optimalkan faktor-faktor tersebut.
* Uji A/B Testing: Lakukan uji A/B testing pada elemen-elemen email (subject line, konten, CTA) untuk melihat variasi mana yang menghasilkan conversion rate yang lebih tinggi.
